drag/drag.view.ts

```diff
@@ -1,41 +1,181 @@
+        }
+  
+        setTimeout(() => this.status("drag"));
+  
+        this.dragged_task = event.target as HTMLElement;
+        this.initial_task_index = Array.from(
+          this.dragged_task.parentElement!.children
+        ).indexOf(this.dragged_task);
+  
+        const transfer = this.transfer();
+        for (let type in transfer) {
+          (event as DragEvent).dataTransfer!.setData(
+            type,
+            transfer[type as keyof typeof transfer]
+          );
+        }
+  
+        (event as DragEvent).dataTransfer!.setDragImage(this.image(), 0, -32);
+  
+        const effects = [] as string[];
+        if (this.allow_copy()) effects.push("Copy");
+        if (this.allow_link()) effects.push("Link");
+        if (this.allow_move()) effects.push("Move");
+  
+        let effectAllowed = effects[0].toLowerCase() + effects.slice(1).join("");
+        if (effectAllowed === "copyLinkMove") effectAllowed = "all";
+        (event as DragEvent).dataTransfer!.effectAllowed =
+          effectAllowed as DataTransfer["effectAllowed"];
+  
+        this.start(event as DragEvent);
+      }
+  
+      drag_move(event: DragEvent | TouchEvent) {
+        if (event instanceof TouchEvent) {
+          this.handle_touch_move(event);
+          return;
+        }
+  
+        event.preventDefault();
+  
+        const draggedOverElem = document.elementFromPoint(
+          event.clientX,
+          event.clientY
+        );
+        if (
+          draggedOverElem &&
+          draggedOverElem !== this.dragged_task &&
+          draggedOverElem.parentElement === this.dragged_task!.parentElement
+        ) {
+          const parent = this.dragged_task!.parentElement;
+          const draggedOverIndex = Array.from(parent!.children).indexOf(
+            draggedOverElem as HTMLElement
+          );
+  
+          if (draggedOverIndex !== this.initial_task_index) {
+            parent!.insertBefore(this.dragged_task!, draggedOverElem);
+            this.initial_task_index = draggedOverIndex;
+          }
+        }
+      }
+  
+      drag_end(event: DragEvent | TouchEvent) {
+        if (event instanceof TouchEvent) {
+          this.handle_touch_end(event);
+          return;
+        }
+  
+        setTimeout(() => this.status("ready"));
+  
+        this.dragged_task = null;
+        this.initial_task_index = null;
+  
+        this.end(event as DragEvent);
+      }
+  
+      handle_touch_start(event: TouchEvent) {
+        event.preventDefault();
+        this.status("drag");
+  
+        this.dragged_task = event.target as HTMLElement;
+        this.initial_task_index = Array.from(
+          this.dragged_task!.parentElement!.children
+        ).indexOf(this.dragged_task);
+  
+        this.dragImageElem = this.dragged_task.cloneNode(true) as HTMLElement;
+  
+        this.dragImageElem.style.position = "absolute";
+        this.dragImageElem.style.pointerEvents = "none"; 
+        this.dragImageElem.style.zIndex = "9999";
```

Comment:
`dragImageElem` should be located in `drag.view.tree`.
Static styles of it in `drag.view.css.ts`, dynamic styles in view.tree `style *`
Name should be in snake_case.

Also I suggest not to clone the dragged element, it may be large and this may cause glitches since the cloned elements will have the same ids.
Instead I suggest making `Drag_label $mol_paragraph` which will just be the title of the dragged element